src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -2591,6 +2591,45 @@ module Queries = struct
             (* Prefix string to disambiguate *)
             "other network: " ^ s )
 
+  let protocol_state =
+    io_field "protocolState"
+      ~doc:"Get the current protocol state, optionally encoded in Base64"
+      ~typ:(non_null string)
+      ~args:
+        Arg.
+          [ arg "encoding" ~doc:"Encoding format (JSON or BASE64)"
+              ~typ:
+                (enum "Encoding"
+                   ~values:
+                     [ enum_value "JSON" ~value:`JSON
+                     ; enum_value "BASE64" ~value:`BASE64
+                     ] )
+          ]
+      ~resolve:(fun { ctx = mina; _ } () encoding_opt ->
+        match Mina_lib.best_tip mina with
+        | `Active best_tip ->
+            let protocol_state =
+              Transition_frontier.Breadcrumb.protocol_state best_tip
+            in
+            let encoded =
+              match encoding_opt with
+              | Some `JSON ->
+                  Mina_state.Protocol_state.value_to_yojson protocol_state
+                  |> Yojson.Safe.to_string
+              | Some `BASE64 ->
+                  Bin_prot.Writer.to_string
+                    Mina_state.Protocol_state.Value.Stable.V2.bin_t.writer
+                    protocol_state
+                  |> Base64.encode_exn
+              | None ->
+                  (* Default to JSON if no encoding is specified *)
+                  Mina_state.Protocol_state.value_to_yojson protocol_state
+                  |> Yojson.Safe.to_string
```

Comment:
Minor enhancement: join the default case with the JSON with an "or-pattern"
```suggestion
              match encoding_opt with
              | Some `BASE64 ->
                  Bin_prot.Writer.to_string
                    Mina_state.Protocol_state.Value.Stable.V2.bin_t.writer
                    protocol_state
                  |> Base64.encode_exn
              | Some `JSON 
              | None ->
                  (* Default to JSON if no encoding is specified *)
                  Mina_state.Protocol_state.value_to_yojson protocol_state
                  |> Yojson.Safe.to_string
```

Reply:
Thanks!